scalapact-core/src/test/scala/com/itv/scalapactcore/common/JsonEqualitySpec.scala

```diff
@@ -151,5 +150,78 @@ class JsonEqualitySpec extends FunSpec with Matchers {
 
   }
 
+  describe("Specific cases") {
+
+    it("should cope with case #1") {
+
+      val a =
+        """
+          |{
+          |  "x": {
+          |    "y": [
+          |      {
+          |        "myDateTime": {
+          |          "at": "2016-12-18T21:00Z"
+          |        }
+          |      },
+          |      {
+          |        "myDateTime": {
+          |          "at": "2017-01-11T00:01Z"
+          |        }
+          |      }
+          |    ]
+          |  }
+          |}
+        """.stripMargin
+
+      val b =
+        """
+          |{
+          |  "x": {
+          |    "y": [
+          |      {
+          |        "myDateTime": {
+          |          "at": "2016-12-18T21:00Z"
+          |        }
+          |      },
+          |      {
+          |        "myDateTime": {
+          |          "at": "2017-01-11T00:01Z"
+          |        }
+          |      }
+          |    ]
+          |  }
+          |}
+        """.stripMargin
+
+      val regex = "^(\\d{4})-(\\d{2})-(\\d{2})T(\\d{2})(:(\\d{2})){0,1}(Z|\\+0100)$"
+
+      val rules: Option[Map[String, MatchingRule]] =
+        Option(
+          Map(
+            "$.body.x.y[*].myDateTime.at" -> MatchingRule(Option("regex"), Option(regex), None)
+          )
+        )
+
+      withClue("Empty A") {
+        ("{}".parseOption.get =~ b.parseOption.get) (rules) shouldEqual true
+      }
```

Comment:
This is a red herring. You might think (as I did and hey, I wrote it...) that the rules are only checked against the received data - and they are - but ONLY if you defined the field in your example "expected" json. Therefore, this test always passes.